### PR TITLE
Auto-download missing objects

### PR DESCRIPTION
**Pull request by IntelOrca**
This adds a new button to the missing object window to automatically download the missing objects.

Sometimes I get a chain of 502 errors when querying for objects I am wondering whether that is me, curl or the API. Would anyone be able to test this and let me know if they hit the same issue?

This downloads objects synchronously, this can be improved but I don't want to over-complicate the initial feature PR.

image

### CODE REVIEW

**Comment by janisozaur** [comment]
@IntelOrca can you update your x86-64 compiler configuration to enable the warning all other CI jobs raised? (Comparison of signed to unsigned values.)

**Comment by AaronVanGeffen** [comment]
This is really neat feature. Thanks for working on it.

Just for the record, what does the repository on our end consist of? We should make sure not to inadvertently distribute any copyrighted (official) objects, of course.

**Comment by IntelOrca** [comment]
@AaronVanGeffen It is an endpoint that returns a URL of where to download the object, currently NEDesigns and they do not allow you to download the vanilla objects.

**Comment by ZehMatt** [comment]
This seems cool but how would I actually test it?

**Comment by telk5093** [comment]
Nice feature, seems like online contents download of OpenTTD.
btw is there any plan to _upload_ objects like BaNaNaS of OpenTTD?

**Comment by Gymnasiast** [comment]
>  btw is there any plan to _upload_ objects like BaNaNaS of OpenTTD?

Ultimately, yes.

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -3576,7 +3576,7 @@ STR_6125    :Object type
 STR_6126    :Unknown type
 STR_6127    :File: {STRING}
 STR_6128    :The file could not be loaded as some of the objects referenced in it are missing or corrupt. A list of these objects is given below.
-STR_6129    :Copy selected item to clipboard
+STR_6129    :Copy item to clipboard
 STR_6130    :Copy entire list to clipboard
```
Comment (AaronVanGeffen) [inline data/language/en-GB.txt]: Perhaps 'Copy entire list to clipboard' should be shortened as well. It barely fits in English, but I think both Dutch and German will be in trouble for this string, too.

Comment (Gymnasiast) [reply data/language/en-GB.txt]: Shortening it in English does not solve any problems for Dutch and English. Texts should not be translated literally, they should make the idea behind the message clear. (Also, AFAIK, the current Dutch does fit.)

Comment (AaronVanGeffen) [reply data/language/en-GB.txt]: I don't disagree, but the localisation aspect was not the main point I was trying to make. The three buttons are equally wide, but their strings are not — especially with two being overly full. I'm hoping to get some balance back.

Comment (Gymnasiast) [reply data/language/en-GB.txt]: Perhaps we should change the buttons to `Copy`, `Copy all` and `Download all`, then, with the current button texts as tooltips.

Comment (IntelOrca) [reply data/language/en-GB.txt]: Done

**Pull request merged**